normandy/recipes/tests/test_exports.py

```diff
@@ -0,0 +1,187 @@
+        settings.REMOTE_SETTINGS_USERNAME = None
+        settings.REMOTE_SETTINGS_PASSWORD = self.test_settings["PASSWORD"]
+        with pytest.raises(ImproperlyConfigured) as exc:
+            exports.RemoteSettings()
+        assert "REMOTE_SETTINGS_USERNAME" in str(exc)
+
+        # Leave out PASSWORD
+        settings.REMOTE_SETTINGS_URL = self.test_settings["URL"]
+        settings.REMOTE_SETTINGS_USERNAME = self.test_settings["USERNAME"]
+        settings.REMOTE_SETTINGS_PASSWORD = None
+        with pytest.raises(ImproperlyConfigured) as exc:
+            exports.RemoteSettings()
+        assert "REMOTE_SETTINGS_PASSWORD" in str(exc)
+
+        # Leave out COLLECTION_ID
+        settings.REMOTE_SETTINGS_URL = self.test_settings["URL"]
+        settings.REMOTE_SETTINGS_USERNAME = self.test_settings["USERNAME"]
+        settings.REMOTE_SETTINGS_PASSWORD = self.test_settings["PASSWORD"]
+        settings.REMOTE_SETTINGS_COLLECTION_ID = None
+        with pytest.raises(ImproperlyConfigured) as exc:
+            exports.RemoteSettings()
+        assert "REMOTE_SETTINGS_COLLECTION_ID" in str(exc)
+
+        # Include everything
+        settings.REMOTE_SETTINGS_URL = self.test_settings["URL"]
+        settings.REMOTE_SETTINGS_USERNAME = self.test_settings["USERNAME"]
+        settings.REMOTE_SETTINGS_PASSWORD = self.test_settings["PASSWORD"]
+        settings.REMOTE_SETTINGS_COLLECTION_ID = "normandy-recipes"
+        # assert doesn't raise
+        exports.RemoteSettings()
+
+    def test_publish_puts_record_and_approves(self, settings, mock_logger, mock_kinto_request):
+        """Test that requests are sent to Remote Settings on publish."""
+
+        settings.REMOTE_SETTINGS_URL = self.test_settings["URL"]
+        settings.REMOTE_SETTINGS_USERNAME = self.test_settings["USERNAME"]
+        settings.REMOTE_SETTINGS_PASSWORD = self.test_settings["PASSWORD"]
+
+        mock_kinto_request.return_value = mock.MagicMock(status_code=200)
+        recipe = RecipeFactory(name="Test", approver=UserFactory(), enabler=UserFactory())
+
+        remotesettings = exports.RemoteSettings()
+        remotesettings.publish(recipe)
+
+        auth = (settings.REMOTE_SETTINGS_USERNAME, settings.REMOTE_SETTINGS_PASSWORD)
+        headers = {"User-Agent": KINTO_USER_AGENT, "Content-Type": "application/json"}
+        url = settings.REMOTE_SETTINGS_URL + "/buckets/main-workspace/collections/normandy-recipes"
+        record_url = url + "/records/" + str(recipe.id)
+
+        # Assert the correct request was made
+        payload = json.dumps({"data": exports.recipe_as_record(recipe)})
+
+        assert mock_kinto_request.call_count == 2
+        mock_kinto_request.assert_any_call(
+            "put", record_url, data=payload, auth=auth, headers=headers
+        )
+        mock_kinto_request.assert_any_call(
+            "patch", url, data='{"data": {"status": "to-sign"}}', auth=auth, headers=headers
+        )
```

Comment:
Perhaps we should be using `requestsmock` since the `requests` that `kinto_client` uses is the same one.

Reply:
Done :)